### ISSUE TITLE
ui.window.locale should be set on startup

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#1865]
Currently, it's not, which means we waste 16ms in every app with a MaterialApp because we spin an entire frame waiting for the locale to be set.

This requires some redundant pumps in tests, too.

**Comment by HansMuller** [comment]
The current version of this problem, once https://github.com/flutter/flutter/pull/11497 lands, is that if we have to wait for an app's async localizations to load, then we're going to at least miss the "first" frame. More than one frame if localizations loading actually takes longer than 16ms.

Apps could create initial screens with more than one Localizations widget. That means that we a way for more than one async loader to defer the "first frame".

Maybe WidgetsApp should create a  DeferFirstFrame inherited widget. Localizations widgets would use DeferFirstFrame.push() when an async load started and DeferFirstFrame.pop()  when the load finished.

**Comment by Hixie** [comment]
I wouldn't use a widget for this. Just something like binding.lockInput().

**Comment by Piinks** [comment]
Closing this as resolved. The `ui.window.locale` (now `ViewConfiguration.locale` in the modern framework) is properly initialized on startup. Modern Flutter applications have robust ways to access and react to the system locale from the start of the application lifecycle via `WidgetsBinding.instance.platformDispatcher.locale` or through `MaterialApp`/`CupertinoApp`'s built-in localization delegates.

If there is more work to do here, please let us know by filing a new issue with up to date information. Thanks!
